app/infrastructure/services/webhook_dispatcher.py

```
@@ -0,0 +1,161 @@
+"""Webhook dispatcher: POST events to subscribed URLs with HMAC signing and retries."""
+
+from __future__ import annotations
+
+import asyncio
+import hashlib
+import hmac
+import json
+import logging
+from typing import TYPE_CHECKING, Any, Awaitable, Callable
```

Ruff (0.15.5)

[warning] Import from `collections.abc` instead: `Awaitable`, `Callable`

Import from `collections.abc`

(UP035)